File: hangingout/c/hangingout.c
#include <stdio.h>

int main()
{
  int limit = 0, events = 0, curr = 0, people = 0, rejected = 0;
  scanf("%d %d", &limit, &events);
  for (int i = 0; i < events; i++)
  {
    char event[6];
    scanf("%s %d", event, &people);
    if (event[0] == 'e')
    {
      if (curr + people > limit)
      {
        rejected++;
      } else {
        curr += people;
      }
    } else {
      curr -= people;
    }
  }
  printf("%d\n", rejected);
  return 0;
}

File: hangingout/py/hangingout.py
limit, events = map(int, input().split())
current = 0
count = 0
for i in range(events):
  event = input().split()
  people = int(event[1])
  if event[0] == "enter":
    if current + people > limit:
      count += 1
    else:
      current += people
  else:
    current -= people
print(count)
